google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -188,12 +199,25 @@ def __init__(
                 '"[a-z][-a-z0-9]{{0,127}}"'.format(job_id)
             )
 
-        builder = pipeline_utils.PipelineRuntimeConfigBuilder.from_job_spec_json(
-            pipeline_job
-        )
-        builder.update_pipeline_root(pipeline_root)
-        builder.update_runtime_parameters(parameter_values)
-        runtime_config_dict = builder.build()
+        if pipeline_spec_only:
```

Comment:
Sorry I wasn't very clear with the last comment, I meant you could use `PipelineRuntimeConfigBuilder.from_job_spec_json(pipeline_job)` for both cases, something like:

```
if pipeline_json.get("pipelineSpec") is not None:
    pipeline_job = pipeline_json
    pipeline_root = ...
else:
    pipeline_job = {"pipelineSpec": pipeline_json, "runtimeConfig": {}}
    pipeline_root = ...

builder = pipeline_utils.PipelineRuntimeConfigBuilder.from_job_spec_json(pipeline_job)
builder.update_pipeline_root(pipeline_root)
builder.update_runtime_parameters(parameter_values)
runtime_config_dict = builder.build()
```

This will result shorter and cleaner code.